pageserver/src/pgdatadir_mapping.rs

```diff
@@ -1069,6 +1096,28 @@ impl<'a> DatadirModification<'a> {
         Ok(())
     }
 
+    pub async fn set_replorigin(
+        &mut self,
+        origin_id: RepOriginId,
+        origin_lsn: Lsn,
+    ) -> anyhow::Result<()> {
+        let mut repl_origins = self.tline.repl_origins.lock().await;
+        let origins = repl_origins.get_or_insert_with(ReplOrigins::default);
+        origins.map.insert(origin_id, origin_lsn);
+        let buf = ReplOrigins::ser(origins)?;
```

Comment:
If the `self.tline.repl_origins` cache isn't loaded yet, the `get_or_insert_with(ReplOrigins::default)` call will replace it with an empty value, losing all the other replication origins.

(With the approach of separate key-value pair for each replication origin, we don't need the cache at all, so this will become moot)